### PR TITLE
Fix Header Option Casting

### LINKED ISSUE #257: Null error in chopper_generator 4.0.0
Since upgrading `chopper_generator` to 4.0.0 I get this error when running the build with the `--verbose` option:
```
[SEVERE] chopper_generator:chopper_generator:

type 'Null' is not a subtype of type 'String'
package:chopper_generator/chopper_generator.dart 7:58                   chopperGeneratorFactory
```

The erroring code is this:
```dart
library chopper_generator.dart;

import 'package:build/build.dart';
import 'src/generator.dart';

Builder chopperGeneratorFactory(BuilderOptions options) =>
    chopperGeneratorFactoryBuilder(header: options.config['header'] as String);
```

Changing it to this:
```dart
Builder chopperGeneratorFactory(BuilderOptions options) =>
    chopperGeneratorFactoryBuilder(header: options.config['header'] ?? '');
```
fixes the issue. Everything built just fine after that.

### PR DESCRIPTION
**Pull request by YoussefRaafatNasry**
fixes #257

### CODE REVIEW

**Review by JEuler — APPROVED** [review]

**Comment by vip20** [comment]
When will this be live? Breaks my code from building.

**Comment by alfredo-handcash** [comment]
@JEuler Can you merge it? Thank you!

File: chopper_generator/lib/src/generator.dart [hunk 1]
```diff
@@ -503,10 +503,10 @@ class ChopperGenerator extends GeneratorForAnnotation<chopper.ChopperApi> {
   }
 }
 
-Builder chopperGeneratorFactoryBuilder({String header = ''}) => PartBuilder(
+Builder chopperGeneratorFactoryBuilder({String? header}) => PartBuilder(
       [ChopperGenerator()],
       '.chopper.dart',
-      header: header,
+      header: header ?? '',
```
Comment (JEuler) [inline chopper_generator/lib/src/generator.dart:509]: I think we should pass it without ?? operator. It is supported on inner levels.

**Comment by JEuler** [comment]
One remark for @YoussefRaafatNasry and I will do that. :)

**Review by JEuler — APPROVED** [review]

**Pull request merged**